### PR TITLE
SN4800 xcvrd changes

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**

### CODE REVIEW

File: sonic-xcvrd/xcvrd/xcvrd.py [hunk 1]
```diff
@@ -89,26 +92,11 @@
...
-            return None
-
 # Get physical port name
 
 
 def get_physical_port_name(logical_port, physical_port, ganged):
-    if logical_port == physical_port:
-        return logical_port
-    elif ganged:
+    if ganged:
```
Comment (Junchao-Mellanox) [inline sonic-xcvrd/xcvrd/xcvrd.py:106]: remove ganged

File: sonic-xcvrd/xcvrd/xcvrd.py [hunk 2]
```diff
@@ -848,40 +795,32 @@ def init_port_sfp_status_tbl(stop_event=threading.Event()):
...
 
-                if not sfp_status_helper.detect_port_in_error_status(logical_port_name, status_tbl[asic_index]):
-                    post_port_dom_info_to_db(logical_port_name, dom_tbl[asic_index], self.task_stopping_event)
-                    post_port_dom_threshold_info_to_db(logical_port_name, dom_tbl[asic_index], self.task_stopping_event)
+                if not sfp_status_helper.detect_port_in_error_status(logical_port_name, xcvr_table_helper.get_status_tbl(asic_index)):
+                    post_port_dom_info_to_db(logical_port_name, self.port_mapping, xcvr_table_helper.get_dom_tbl(asic_index), self.task_stopping_event)
+                    post_port_dom_threshold_info_to_db(logical_port_name, self.port_mapping, xcvr_table_helper.get_dom_tbl(asic_index), self.task_stopping_event)
                     if y_cable_presence[0] is True:
-                        y_cable_helper.check_identifier_presence_and_update_mux_info_entry(state_db, mux_tbl, asic_index, logical_port_name)
+                        y_cable_helper.check_identifier_presence_and_update_mux_info_entry(xcvr_table_helper.get_state_db(), mux_tbl, asic_index, logical_port_name)
```
Comment (Junchao-Mellanox) [inline sonic-xcvrd/xcvrd/xcvrd.py]: Need pass asic_index to get_state_db()

File: sonic-xcvrd/xcvrd/xcvrd.py [hunk 3]
```diff
@@ -896,13 +835,45 @@ def task_stop(self):
...
 # Process wrapper class to update sfp state info periodically
 
 
 class SfpStateUpdateTask(object):
-    def __init__(self):
+    RETRY_EEPROM_READING_INTERVAL = 60
+    def __init__(self, port_mapping):
         self.task_process = None
         self.task_stopping_event = multiprocessing.Event()
+        self.event_queue = multiprocessing.Queue()
```
Comment (Junchao-Mellanox) [inline sonic-xcvrd/xcvrd/xcvrd.py:924]: maybe remove?

File: sonic-xcvrd/xcvrd/xcvrd.py [hunk 4]
```diff
@@ -1173,13 +1145,154 @@ def task_run(self, sfp_error_event, y_cable_presence):
...
+        # The operation to remove entry from TRANSCEIVER_DOM_INFO is duplicate with DomInfoUpdateTask.on_remove_logical_port,
+        # but it is necessary because TRANSCEIVER_DOM_INFO is also updated in this sub process when a new SFP is inserted.
+        del_port_sfp_dom_info_from_db(port_change_event.port_name, 
+                                      self.port_mapping, 
+                                      xcvr_table_helper.get_int_tbl(port_change_event.asic_id), 
+                                      xcvr_table_helper.get_dom_tbl(port_change_event.asic_id))
+        delete_port_from_status_table(port_change_event.port_name, xcvr_table_helper.get_status_tbl(port_change_event.asic_id))
+
+        if port_change_event.port_name in self.retry_eeprom_set:
+            self.retry_eeprom_set.remove(port_change_event.port_name)
```
Comment (Junchao-Mellanox) [inline sonic-xcvrd/xcvrd/xcvrd.py:1270]: probably need remove sfp error table

File: sonic-xcvrd/xcvrd/xcvrd.py [hunk 5]
```diff
@@ -1173,13 +1145,154 @@ def task_run(self, sfp_error_event, y_cable_presence):
...
+
+    def on_add_logical_port(self, port_change_event):
+        # A logical port is created. There could be 3 cases:
+        #  1. SFP information is already in DB, which means that a logical port with the same physical index is in DB before.
```
Comment (Junchao-Mellanox) [inline sonic-xcvrd/xcvrd/xcvrd.py:1282]: No need query EEPROM in update task if SFP information is already in DB, we can copy here.

File: sonic-xcvrd/xcvrd/xcvrd.py [hunk 6]
```diff
@@ -1173,13 +1145,154 @@ def task_run(self, sfp_error_event, y_cable_presence):
...
+                    if error_dict:
+                        vendor_specific_error_description = error_dict.get(port_change_event.port_index)
+                    else:
+                        vendor_specific_error_description = _wrapper_get_sfp_error_description(port_change_event.port_index)
+                    error_descriptions.append(vendor_specific_error_description)
+
+                error_description = '|'.join(error_descriptions)
+                helper_logger.log_info("Receive error update port sfp status table.")
+                if sfp_status_helper.is_error_block_eeprom_reading(error_bits):
+                    read_eeprom = False
```
Comment (Junchao-Mellanox) [inline sonic-xcvrd/xcvrd/xcvrd.py:1343]: init read_eeprom?

File: sonic-xcvrd/xcvrd/xcvrd.py (hunk 3 above)
Comment (Junchao-Mellanox) [reply sonic-xcvrd/xcvrd/xcvrd.py:924]: no, need this to do inter-process communication

File: sonic-xcvrd/xcvrd/xcvrd.py (hunk 6 above)
Comment (Junchao-Mellanox) [reply sonic-xcvrd/xcvrd/xcvrd.py:1343]: Already inited

File: sonic-xcvrd/xcvrd/xcvrd.py (hunk 4 above)
Comment (Junchao-Mellanox) [reply sonic-xcvrd/xcvrd/xcvrd.py:1270]: SFP error table is only remove when SFP event changed to insert or remove, so no need to remove it here.

**Review by keboliu — APPROVED** [review]

**Pull request closed without merging**